File: codigo/backend/go-backend/models/estoque.go
package models

import "fmt"


type Estoque struct {
    LojaID    uint    `gorm:"primaryKey"`
    ProdutoID uint    `gorm:"primaryKey"`
    Quantidade int     `gorm:"not null"`
    Preco     float64 `gorm:"type:decimal(10,2);not null"`
    Loja      Loja    `gorm:"foreignKey:LojaID;references:ID"`
    Produto   Produto `gorm:"foreignKey:ProdutoID;references:ID"`
}


func (Estoque) TableName() string {
    return "estoques" // Nome da tabela no banco de dados
}

func (e *Estoque) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "loja_id":    e.LojaID,
        "produto_id": e.ProdutoID,
        "quantidade": e.Quantidade,
        "preco":      e.Preco,
    }
}

func (e *Estoque) String() string {
    return fmt.Sprintf("<Estoque %d-%d>", e.LojaID, e.ProdutoID)
}

File: codigo/backend/sap-ecc-emulator/src/models/estoque.py
from db import db

class Estoque(db.Model):
    __tablename__ = 'estoques'

    loja_id = db.Column(db.Integer, db.ForeignKey('lojas.id'), primary_key=True)
    produto_id = db.Column(db.Integer, db.ForeignKey('produtos.id'), primary_key=True)
    quantidade = db.Column(db.Integer, nullable=False)
    preco = db.Column(db.Numeric(10, 2), nullable=False)

    loja = db.relationship('Loja', backref='estoques', lazy=True)
    produto = db.relationship('Produto', backref='estoques', lazy=True)

    def to_dict(self):
        return {
            'loja_id': self.loja_id,
            'produto_id': self.produto_id,
            'quantidade': self.quantidade,
            'preco': float(self.preco)  # Convert Numeric to float for JSON serialization
        }

    def __repr__(self):
        return f'<Estoque {self.loja_id}-{self.produto_id}>'
